File: src/main/java/fun/spmc/smpmod/misc/ItemRarity.java
package fun.spmc.smpmod.misc;

import net.minecraft.ChatFormatting;

public enum ItemRarity {
    COMMON(ChatFormatting.WHITE, false),
    UNCOMMON(ChatFormatting.GREEN, false),
    RARE(ChatFormatting.BLUE, false),
    EPIC(ChatFormatting.DARK_PURPLE, false),
    LEGENDARY(ChatFormatting.GOLD, true),
    MYTHIC(ChatFormatting.LIGHT_PURPLE, true),
    CHROMATIC(ChatFormatting.RED, true),
    CELESTIAL(ChatFormatting.AQUA, true);

    private final ChatFormatting color;
    private final boolean consideredRare;

    ItemRarity(ChatFormatting color, boolean consideredRare) { this.color = color; this.consideredRare = consideredRare; }
    public ChatFormatting getColor() { return color; }
    public boolean shouldAnnounce() { return consideredRare; }
    public String toString() { return name().charAt(0) + name().substring(1).toLowerCase(); }
}

File: src/main/java/fun/spmc/smpmod/core/ItemRarity.kt
package `fun`.spmc.smpmod.core

import net.minecraft.network.chat.TextColor
import java.util.*

enum class ItemRarity(@JvmField val color: TextColor, private val consideredRare: Boolean) {
    COMMON(TextColor.fromRgb(10529458), false),
    UNCOMMON(TextColor.fromRgb(3066993), false),
    RARE(TextColor.fromRgb(3377407), false),
    EPIC(TextColor.fromRgb(10837738), false),
    LEGENDARY(TextColor.fromRgb(16757058), true),
    MYTHIC(TextColor.fromRgb(16740518), true),
    CHROMATIC(TextColor.fromRgb(16726072), true),
    ASTRAL(TextColor.fromRgb(4581119), true);

    fun shouldAnnounce(): Boolean { return consideredRare }
    override fun toString(): String { return name[0].toString() + name.substring(1).lowercase(Locale.getDefault()) }
}
